test/test_charm.py

```diff
@@ -473,6 +476,46 @@ def _get_action_test_meta(cls):
   additionalProperties: false
 ''')
 
+    def test_meta_from_charm_root(self):
+        with tempfile.TemporaryDirectory() as d:
+            td = pathlib.Path(d)
+            (td / 'metadata.yaml').write_text(
+                yaml.safe_dump(
+                    {"name": "bob",
+                        "requires": {
+                            "foo":
+                                {"interface": "bar"}
+                        }}))
```

Comment:
The formatting is very odd here. Can we clean up, or just write text directly? Similar below.

```suggestion
            (td / 'metadata.yaml').write_text("""
name: bob
requires:
  foo:
    interface: bar
""")
                yaml.safe_dump(
                    {"name": "bob",
                        "requires": {
                            "foo":
                                {"interface": "bar"}
                        }}))
```